src/library/vicon_estimator.cpp

```diff
@@ -398,14 +434,24 @@ bool RotationalEstimator::detectMeasurementOutlier(const Eigen::Quaterniond& ori
     return false;
   }
 
-  // Constructing the quaternion representing the body to measurement rotation
-  Eigen::Quaterniond rotation_subsequent_measurements = orientation_measured
-      * orientation_measured_old_.inverse();
-  // Calculating a measure of the size of the error quaternion
-  double rotation_magnitude_radians = quaternionRotationMagnitude(rotation_subsequent_measurements);
+  // Constructing the quaternion representing the rotation between subsquent measurements
+  Eigen::Quaterniond q_Z_Z1_ = orientation_measured * orientation_measured_old_.inverse();
+  // Calculating the quaternion magnitude
+  double q_Z_Z1_magnitude_ = quaternionRotationMagnitude(q_Z_Z1_);
+
+  // Constructing the quaternion representing the rotation between the body a measurement
+  Eigen::Quaterniond q_Z_B_ = orientation_measured * orientation_estimate_B_W_.inverse();
+  // Calculating the quaternion magnitude
+  double q_Z_B_magnitude_ = quaternionRotationMagnitude(q_Z_B_);
+
+  // Writing the error quaternions and their magnitude to the intermediate results structure
+  estimator_results_.q_Z_Z1_ = q_Z_Z1_;
+  estimator_results_.q_Z_Z1_magnitude_ = q_Z_Z1_magnitude_;
```

Comment:
not sure if we need to store the magnitude, since this information is easy to retrieve from the quaternion..